src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -156,6 +156,25 @@ public ResponseEntity<Set<Lecture>> getLecturesForCourse(@PathVariable Long cour
         return ResponseEntity.ok().body(lectures);
     }
 
+    /**
+     * GET /courses/:courseId/lectures : get all the lectures of a course with their lecture units and slides
+     *
+     * @param courseId the courseId of the course for which all lectures should be returned
+     * @return the ResponseEntity with status 200 (OK) and the set of lectures in body
+     */
+    @GetMapping(value = "/courses/{courseId}/lectures-with-slides")
+    @PreAuthorize("hasRole('USER')")
```

Comment:
I don't think it makes sense to create that new endpoints. They are both accessible by Users and I don't think returning the additional data does create that much overhead, or does it?

We could maybe even extend the normal retrieval endpoint if that's the case.

Reply:
I think it creates overhead because we dont need the slide objects at all in other places and we might have many lectures and those lectures might have many units and units will have at list 20-30 slides so why having all those objects across Artemis when we will not use? 

IMO we should have new endpoint

Reply:
I hope you don't mind; I'd like to have @krusche's opinion on this:
What's realistic here. What is preferred in your opinion?